src/main/java/org/jetbrains/bsp/bazel/server/BepServer.java

```diff
@@ -178,6 +143,12 @@ private void processedStartedEvent(BuildEventStreamProtos.BuildStarted buildStar
     taskParkingLot.add(taskId);
   }
 
+  private void processFinishedEventIfPresent(BuildEventStreamProtos.BuildEvent event) {
+    if (event.hasFinished()) {
+      processFinishedEvent(event.getFinished());
```

Comment:
You mean just moving the body of ``processFinishedEvent`` into ``if (event.hasFinished()) {``?

Reply:
yes

Reply:
I disagree - this logic is too complex for being inside the `if` statement - it should be separated to the method. Nevertheless I would consider better names - `processFinishedEvent` and `processFinishedEventForFinishedEvent` or something similar to show that this method should be called only for the finished event.

Reply:
Maybe I'd suggest using `processEvent` for processing if they're present or not and `consumeEvent` for the actual logic of the event. Wdyt? I think consuming already underlies that it must be present.

Reply:
I also think unsplitting it is a bad idea, it's much harder to read code with such nested ifs and easier if you have a properly named method. As to the names, I used ``*ifPresent`` because for each type of event we check if ``event`` _has_ it. I think it's more obvious from outside what happens here. But André's idea is also fine, so I can change it to this if you want. It would be like this, right?:

```
 private void processActionEvent(BuildEventStreamProtos.BuildEvent event) {
    if (event.hasAction()) {
      consumeActionEvent(event.getAction());
    }
  }
```